### PR TITLE
:bug: fix: validate rules shift errors

### PR DESCRIPTION
**Pull request by jardel0x03**
- fix issue #11793

### CODE REVIEW

**Comment by JardelCheung** [comment]
test branch
https://github.com/JardelCheung/react-hook-form/tree/test/issue-11793
test video
https://github.com/react-hook-form/react-hook-form/assets/52777763/c2f44236-f581-4a80-b57f-f4fbe8dc70b0

**Comment by bluebill1049** [comment]
thanks for the fix could you include a test case for this issue?

**Comment by JardelCheung** [comment]
@bluebill1049 I added a test about this issue

File: src/logic/updateFieldArrayRootError.ts [hunk 1]
```diff
@@ -13,7 +13,7 @@ export default <T extends FieldValues = FieldValues>(
   error: Partial<Record<string, FieldError>>,
   name: InternalFieldName,
 ): FieldErrors<T> => {
-  const fieldArrayErrors = compact(get(errors, name));
+  const fieldArrayErrors = convertToArrayPayload(get(errors, name));
```
[the remark this answers is not in the document]
Comment (JardelCheung) [reply src/logic/updateFieldArrayRootError.ts:16]: `compact` has `.filter(Boolean)` will clear `undefined `or `null`

[the remark this answers is not in the document]
Comment (bluebill1049) [reply src/logic/updateFieldArrayRootError.ts:16]: oh i see that's the root cause of the issue. thanks

**Review by bluebill1049 — APPROVED** [review]

**Pull request merged**